### DISCUSSION TITLE
Checking overlap on different layers?

### DISCUSSION THREAD

**Discussion by joshuacurtiss** [replit/kaboom#165] [Q&A] [answered]
Quick question, is there an easy built-in way to check if two items overlap when they're on different layers?  Basically I'm trying to simulate the flying fish like in Mario Bros, so they need body physics but shouldn't be impacted by other things in the player's layer but should kill the player. But my understanding is the built-in overlap methods (whether `overlapped` event or `isOverlapped()` method) check based on your layer. 

I've got a `getAtPos` function I made that `get()`s all objects and then returns only those that match a given `vec2` point, which works but is not as precise. I guess the math/logic for check a body area is just a little more complex and was making my brain hurt.

Josh

**Comment by lajbel** [comment]
Well, first of all make the fish do not have the `body` component, so that they do not collide with the solid() at the moment, and I think that with that it would be fine, create the homemade jump using `move()` or `moveTo()`

**Reply by joshuacurtiss** [reply]
Yeah I appreciate being able to just implement body and gravity and then tell the fish to jump without having to do my own math for physics. ;-p

**Comment by slmjkdbtl** [comment, marked as answer]
Also removed the diff layer no collision rule in beta and next version

**Reply by joshuacurtiss** [reply]
Thanks @slmjkdbtl I think that's good idea to at least have a parameter that lets you ignore the layers. I imagine in some cases it's useful.
